src/download_manager.cc

```diff
@@ -320,39 +325,16 @@ GroupDownloadManager::RemoveDownloadManager(DownloadManager* dm)
 	}
 }
 
-void
-GroupDownloadManager::PartComplete(bool success)
-{
-	std::unique_lock<std::mutex> lg(access_lock);
-	async_downloads--;
-
-	if (success) {
-		StartNextPart();
-	}
-}
-
-void
-GroupDownloadManager::PartsComplete(bool success, uint32_t n_parts)
-{
-	std::unique_lock<std::mutex> lg(access_lock);
-	async_downloads -= n_parts;
-
-	if (success) {
-		for (uint32_t i = 0; i < n_parts; i++) {
-			StartNextPart();
-		}
-	}
-}
-
 void
 GroupDownloadManager::PartDownloadComplete(GroupDownloadManager* gdm,
 		std::shared_ptr<Aws::Client::AsyncCallerContext> context,
 		bool success)
 {
 	(void) context;
```

Comment:
ntipick:
```suggestion
	std::shared_ptr<const StreamManager::AsyncContext> async_ctx =
		std::static_pointer_cast<const StreamManager::AsyncContext>(context);
	uint64_t download_idx = async_ctx->GetPartN();
	
	std::unique_lock<std::mutex> lg(gdm->access_lock);
	gdm->async_downloads--;
	
	ver("S3 download progress: %u/%u active downloads, part %llu completed %s", 
		gdm->async_downloads, gdm->max_async_downloads, 
		(unsigned long long)download_idx, success ? "successfully" : "with error");
```

How about logging here ? 
```
[VER] S3 download progress: 15/32 active downloads, part 42 completed successfully
[VER] S3 download progress: 14/32 active downloads, part 43 completed with error
```

Reply:
I think something like this isn't a bad idea but I have a few concerns/thoughts... I think debug/verbose logging improvements for asrestore/asbackup should be a separate work item.
1. ver() verbose logging does not work in the c++ files, this is because the global variables used to implement verbose logging `g_verbose` are c11 atomics that c++ doesn't understand. We can work around this by wrapping them but I think that's beyond the scope of this PR.
2. logging here would produce quite a bit of output and we should probably try to associate the log with the object being downloaded. We should think about what we get out of this log and if there is a way to get the same information with less frequent log output, for example, I think checking the file open log lines for associated file closed log lines can tell us when hangs like this exist already, for example, if asrestore hangs without closing certain files, we know that the downloads for those files may be deadlocked.